File: frontend/src/stores/creative.ts
import { defineStore } from 'pinia'
import { ref, computed } from 'vue'
import { api } from '@/utils/api'
import type { 
  CreativeProject,
  DocumentAnalysis,
  MarketAnalysis,
  IdeaGeneration,
  GeneratedImage,
  SolutionDesign 
} from '@/types'

export const useCreativeStore = defineStore('creative', () => {
  // 状态
  const currentProject = ref<CreativeProject | null>(null)
  const documentAnalyses = ref<DocumentAnalysis[]>([])
  const marketAnalyses = ref<MarketAnalysis[]>([])
  const ideas = ref<IdeaGeneration[]>([])
  const images = ref<GeneratedImage[]>([])
  const solution = ref<SolutionDesign | null>(null)
  const loading = ref(false)
  const currentIdea = ref<IdeaGeneration | null>(null)

  // Getters
  const projectStatus = computed(() => currentProject.value?.status || 'draft')
  const hasDocumentAnalysis = computed(() => documentAnalyses.value.length > 0)
  const hasMarketAnalysis = computed(() => marketAnalyses.value.length > 0)

  // Actions
  async function fetchProject(projectId: string) {
    loading.value = true
    try {
      const response = await api.get(`/api/v1/creative/projects/${projectId}`)
      currentProject.value = response.data
    } catch (error) {
      console.error('Failed to fetch project:', error)
    } finally {
      loading.value = false
    }
  }

  async function createDocumentAnalysis(projectId: string, data: any) {
    loading.value = true
    try {
      const response = await api.post(
        `/api/v1/creative/projects/${projectId}/document-analysis/`,
        data
      )
      documentAnalyses.value.push(response.data)
      return response.data
    } catch (error) {
      console.error('Failed to create document analysis:', error)
      throw error
    } finally {
      loading.value = false
    }
  }

  async function createProject(data: Partial<CreativeProject>) {
    loading.value = true
    try {
      const response = await api.post('/api/v1/creative/projects/', data)
      return response.data
    } catch (error) {
      console.error('Failed to create project:', error)
      throw error
    } finally {
      loading.value = false
    }
  }

  async function getMarketAnalyses(projectId: string) {
    loading.value = true
    try {
      const response = await api.get(`/api/v1/creative/projects/${projectId}/market-analysis`)
      return response.data
    } catch (error) {
      console.error('Failed to get market analyses:', error)
      throw error
    } finally {
      loading.value = false
    }
  }

  async function createMarketAnalysis(projectId: string, data: any) {
    loading.value = true
    try {
      const response = await api.post(
        `/api/v1/creative/projects/${projectId}/market-analysis/`,
        data
      )
      marketAnalyses.value.push(response.data)
      return response.data
    } catch (error) {
      console.error('Failed to create market analysis:', error)
      throw error
    } finally {
      loading.value = false
    }
  }

  async function getIdeas(projectId: string) {
    loading.value = true
    try {
      const response = await api.get(`/api/v1/creative/projects/${projectId}/ideas`)
      return response.data
    } catch (error) {
      console.error('Failed to get ideas:', error)
      throw error
    } finally {
      loading.value = false
    }
  }

  async function createIdea(projectId: string, data: any) {
    loading.value = true
    try {
      const response = await api.post(
        `/api/v1/creative/projects/${projectId}/idea-generation/`,
        data
      )
      ideas.value.push(response.data)
      return response.data
    } catch (error) {
      console.error('Failed to create idea:', error)
      throw error
    } finally {
      loading.value = false
    }
  }

  function setCurrentIdea(idea: IdeaGeneration) {
    currentIdea.value = idea
  }

  async function getGeneratedImages(ideaId: string) {
    loading.value = true
    try {
      const response = await api.get(`/api/v1/creative/ideas/${ideaId}/images`)
      return response.data
    } catch (error) {
      console.error('Failed to get generated images:', error)
      throw error
    } finally {
      loading.value = false
    }
  }

  async function createGeneratedImage(projectId: string, ideaId: string, data: any) {
    loading.value = true
    try {
      const response = await api.post(
        `/api/v1/creative/ideas/${ideaId}/images/`,
        data
      )
      images.value.push(response.data)
      return response.data
    } catch (error) {
      console.error('Failed to create generated image:', error)
      throw error
    } finally {
      loading.value = false
    }
  }

  async function getSolution(projectId: string) {
    loading.value = true
    try {
      const response = await api.get(`/api/v1/creative/projects/${projectId}/solution`)
      return response.data
    } catch (error) {
      console.error('Failed to get solution:', error)
      throw error
    } finally {
      loading.value = false
    }
  }

  async function createSolution(projectId: string, data: any) {
    loading.value = true
    try {
      const response = await api.post(
        `/api/v1/creative/projects/${projectId}/solution/`,
        data
      )
      solution.value = response.data
      return response.data
    } catch (error) {
      console.error('Failed to create solution:', error)
      throw error
    } finally {
      loading.value = false
    }
  }

  async function getEvaluation(projectId: string) {
    loading.value = true
    try {
      const response = await api.get(`/api/v1/creative/projects/${projectId}/evaluation`)
      return response.data
    } catch (error) {
      console.error('Failed to get evaluation:', error)
      throw error
    } finally {
      loading.value = false
    }
  }

  async function createEvaluation(projectId: string) {
    loading.value = true
    try {
      const response = await api.post(
        `/api/v1/creative/projects/${projectId}/evaluation/`
      )
      return response.data
    } catch (error) {
      console.error('Failed to create evaluation:', error)
      throw error
    } finally {
      loading.value = false
    }
  }

  async function getExportPreview(projectId: string, options: any) {
    loading.value = true
    try {
      const response = await api.post(
        `/api/v1/creative/projects/${projectId}/export/preview`,
        options
      )
      return response.data
    } catch (error) {
      console.error('Failed to get export preview:', error)
      throw error
    } finally {
      loading.value = false
    }
  }

  async function exportProject(projectId: string, options: any) {
    loading.value = true
    try {
      const response = await api.post(
        `/api/v1/creative/projects/${projectId}/export`,
        options
      )
      return response.data
    } catch (error) {
      console.error('Failed to export project:', error)
      throw error
    } finally {
      loading.value = false
    }
  }

  async function updateProject(projectId: string, data: Partial<CreativeProject>) {
    loading.value = true
    try {
      const response = await api.put(
        `/api/v1/creative/projects/${projectId}`,
        data
      )
      currentProject.value = response.data
      return response.data
    } catch (error) {
      console.error('Failed to update project:', error)
      throw error
    } finally {
      loading.value = false
    }
  }

  async function deleteProject(projectId: string) {
    loading.value = true
    try {
      await api.delete(`/api/v1/creative/projects/${projectId}`)
    } catch (error) {
      console.error('Failed to delete project:', error)
      throw error
    } finally {
      loading.value = false
    }
  }

  return {
    currentProject,
    documentAnalyses,
    marketAnalyses,
    ideas,
    images,
    solution,
    loading,
    projectStatus,
    hasDocumentAnalysis,
    hasMarketAnalysis,
    fetchProject,
    createDocumentAnalysis,
    createProject,
    getMarketAnalyses,
    createMarketAnalysis,
    getIdeas,
    createIdea,
    setCurrentIdea,
    getGeneratedImages,
    createGeneratedImage,
    getSolution,
    createSolution,
    getEvaluation,
    createEvaluation,
    getExportPreview,
    exportProject,
    updateProject,
    deleteProject,
  }
}) 

File: backend/app/api/v1/endpoints/creative.py
from typing import List
from fastapi import APIRouter, Depends, HTTPException
from sqlalchemy.orm import Session
from app import crud, models, schemas
from app.api import deps
from app.utils.pdf_generator import PDFGenerator
from fastapi.responses import FileResponse
import os

router = APIRouter()
pdf_generator = PDFGenerator()

@router.post("/projects/", response_model=schemas.CreativeProject)
def create_project(
    project_in: schemas.CreativeProjectCreate,
    db: Session = Depends(deps.get_db),
    current_user: models.User = Depends(deps.get_current_user)
):
    """创建新项目"""
    project = crud.creative_project.create(db=db, obj_in=project_in)
    return project

@router.get("/projects/", response_model=List[schemas.CreativeProject])
def get_projects(
    db: Session = Depends(deps.get_db),
    skip: int = 0,
    limit: int = 100,
    current_user: models.User = Depends(deps.get_current_user)
):
    """获取项目列表"""
    projects = crud.creative_project.get_by_user(db=db, user_id=current_user.id)
    return projects

@router.post("/projects/{project_id}/document-analysis/", response_model=schemas.DocumentAnalysis)
def create_document_analysis(
    project_id: str,
    analysis_in: schemas.DocumentAnalysisCreate,
    db: Session = Depends(deps.get_db),
    current_user: models.User = Depends(deps.get_current_user)
):
    """创建文档分析"""
    project = crud.creative_project.get(db=db, id=project_id)
    if not project:
        raise HTTPException(status_code=404, detail="Project not found")
    analysis = crud.document_analysis.create(db=db, obj_in=analysis_in)
    return analysis

@router.post("/projects/{project_id}/market-analysis/", response_model=schemas.MarketAnalysis)
def create_market_analysis(
    *,
    db: Session = Depends(deps.get_db),
    project_id: str,
    analysis_in: schemas.MarketAnalysisCreate,
    current_user: models.User = Depends(deps.get_current_user)
):
    """创建市场分析"""
    project = crud.creative_project.get(db=db, id=project_id)
    if not project:
        raise HTTPException(status_code=404, detail="Project not found")
    analysis = crud.market_analysis.create(db=db, obj_in=analysis_in)
    return analysis

@router.post("/projects/{project_id}/idea-generation/", response_model=schemas.IdeaGeneration)
def create_idea_generation(
    *,
    db: Session = Depends(deps.get_db),
    project_id: str,
    idea_in: schemas.IdeaGenerationCreate,
    current_user: models.User = Depends(deps.get_current_user)
):
    """创建创意生成"""
    project = crud.creative_project.get(db=db, id=project_id)
    if not project:
        raise HTTPException(status_code=404, detail="Project not found")
    idea = crud.idea_generation.create(db=db, obj_in=idea_in)
    return idea

@router.post("/ideas/{idea_id}/images/", response_model=schemas.GeneratedImage)
def create_generated_image(
    *,
    db: Session = Depends(deps.get_db),
    idea_id: str,
    image_in: schemas.GeneratedImageCreate,
    current_user: models.User = Depends(deps.get_current_user)
):
    """创建生成图片"""
    idea = crud.idea_generation.get(db=db, id=idea_id)
    if not idea:
        raise HTTPException(status_code=404, detail="Idea not found")
    image = crud.generated_image.create(db=db, obj_in=image_in)
    return image

@router.post("/projects/{project_id}/solution/", response_model=schemas.SolutionDesign)
def create_solution_design(
    *,
    db: Session = Depends(deps.get_db),
    project_id: str,
    solution_in: schemas.SolutionDesignCreate,
    current_user: models.User = Depends(deps.get_current_user)
):
    """创建方案设计"""
    project = crud.creative_project.get(db=db, id=project_id)
    if not project:
        raise HTTPException(status_code=404, detail="Project not found")
    solution = crud.solution_design.create(db=db, obj_in=solution_in)
    return solution

@router.get("/projects/{project_id}", response_model=schemas.CreativeProject)
def get_project(
    *,
    db: Session = Depends(deps.get_db),
    project_id: str,
    current_user: models.User = Depends(deps.get_current_user)
):
    """获取项目详情"""
    project = crud.creative_project.get(db=db, id=project_id)
    if not project:
        raise HTTPException(status_code=404, detail="Project not found")
    return project

@router.get("/projects/{project_id}/document-analysis", response_model=List[schemas.DocumentAnalysis])
def get_document_analyses(
    *,
    db: Session = Depends(deps.get_db),
    project_id: str,
    current_user: models.User = Depends(deps.get_current_user)
):
    """获取文档分析列表"""
    analyses = crud.document_analysis.get_by_project(db=db, project_id=project_id)
    return analyses

@router.get("/projects/{project_id}/market-analysis", response_model=List[schemas.MarketAnalysis])
def get_market_analyses(
    *,
    db: Session = Depends(deps.get_db),
    project_id: str,
    current_user: models.User = Depends(deps.get_current_user)
):
    """获取市场分析列表"""
    analyses = crud.market_analysis.get_by_project(db=db, project_id=project_id)
    return analyses

@router.get("/projects/{project_id}/ideas", response_model=List[schemas.IdeaGeneration])
def get_ideas(
    *,
    db: Session = Depends(deps.get_db),
    project_id: str,
    current_user: models.User = Depends(deps.get_current_user)
):
    """获取创意列表"""
    ideas = crud.idea_generation.get_by_project(db=db, project_id=project_id)
    return ideas

@router.get("/ideas/{idea_id}/images", response_model=List[schemas.GeneratedImage])
def get_generated_images(
    *,
    db: Session = Depends(deps.get_db),
    idea_id: str,
    current_user: models.User = Depends(deps.get_current_user)
):
    """获取生成图片列表"""
    images = crud.generated_image.get_by_idea(db=db, idea_id=idea_id)
    return images

@router.get("/projects/{project_id}/solution", response_model=schemas.SolutionDesign)
def get_solution_design(
    *,
    db: Session = Depends(deps.get_db),
    project_id: str,
    current_user: models.User = Depends(deps.get_current_user)
):
    """获取方案设计"""
    solution = crud.solution_design.get_by_project(db=db, project_id=project_id)
    if not solution:
        raise HTTPException(status_code=404, detail="Solution not found")
    return solution

@router.patch("/projects/{project_id}/status", response_model=schemas.CreativeProject)
def update_project_status(
    *,
    db: Session = Depends(deps.get_db),
    project_id: str,
    status: str,
    current_user: models.User = Depends(deps.get_current_user)
):
    """更新项目状态"""
    project = crud.creative_project.get(db=db, id=project_id)
    if not project:
        raise HTTPException(status_code=404, detail="Project not found")
    
    if status not in ['draft', 'in_progress', 'completed']:
        raise HTTPException(status_code=400, detail="Invalid status")
    
    project_data = {"status": status}
    updated_project = crud.creative_project.update(
        db=db,
        db_obj=project,
        obj_in=project_data
    )
    return updated_project

@router.delete("/projects/{project_id}", status_code=204)
def delete_project(
    *,
    db: Session = Depends(deps.get_db),
    project_id: str,
    current_user: models.User = Depends(deps.get_current_user)
):
    """删除项目"""
    project = crud.creative_project.get(db=db, id=project_id)
    if not project:
        raise HTTPException(status_code=404, detail="Project not found")
    crud.creative_project.remove(db=db, id=project_id)

@router.post("/generate-product-pdf")
async def generate_product_pdf(
    filename: str,
    product_requirements: str,
    product_features: str,
    product_image_url: str,
    marketing_copy: str
):
    try:
        # 生成PDF
        pdf_path = pdf_generator.generate_product_pdf(
            filename=filename,
            product_requirements=product_requirements,
            product_features=product_features,
            product_image_url=product_image_url,
            marketing_copy=marketing_copy
        )
        
        # 返回生成的PDF文件
        return FileResponse(
            pdf_path,
            media_type='application/pdf',
            filename=os.path.basename(pdf_path)
        )
        
    except Exception as e:
        raise HTTPException(status_code=500, detail=str(e))

# 定期清理旧文件的任务
@router.on_event("startup")
async def startup_event():
    pdf_generator.cleanup_old_files()

# 添加其他必要的路由... 